### PR TITLE
Add neovim_bin argument

### LINKED ISSUE #1005: Option to specify path of nvim.exe for neovide to detect.
**Is your feature request related to a problem? Please describe.**
I have to place nvim\bin into path environment variable just so that neovide can sense nvim.exe.

**Describe the solution you'd like**
Add an option to specify nvim path in neovide settings.

### PR DESCRIPTION
**Pull request by LoipesMas**
## What kind of change does this PR introduce?
- Feature

## Did this PR introduce a breaking change? 
- No

Adds command line argument for specifying neovim bin path.
Closes #1005

### CODE REVIEW

**Comment by Kethku** [comment]
This looks good but I would like to test it before merging.

**Comment by Kethku** [comment]
Testing would require:
1. Removing neovim from the path
2. Calling neovide with the bin argument and making sure it works
3. Re-adding neovim to the path
4. Calling neovide with the path to a different neovim install to make sure it works
5. Calling neovide with an invalid neovim path to make sure something useful is messaged

**Comment by LoipesMas** [comment]
```
$ PATH="" neovide
ERROR [neovide::bridge] nvim not found!
$ PATH="" neovide --neovim-bin /usr/bin # this works correctly
$ neovide --neovim-bin /path/to/different/version/nvim # uses the different version (verified by :version)
$ neovide --neovim-bin /incorrect/path
WARN [neovide::bridge] NEOVIM_BIN is invalid falling back to first bin in PATH
# and then launches correctly as if the argument wasn't used
$ PATH="" neovide --neovim-bin /incorrect/path
WARN [neovide::bridge] NEOVIM_BIN is invalid falling back to first bin in PATH
ERROR [neovide::bridge] nvim not found!
```
Also `NEOVIM_BIN` env var overwrites argument. Not sure if this the correct way or should this be the other way around.

**Comment by Kethku** [comment]
This should definitely be the other way around.

File: src/main.rs [hunk 1]
```diff
@@ -194,7 +189,7 @@ pub fn init_logger() {
                 Naming::Timestamps,
                 Cleanup::KeepLogFiles(1),
             ),
-        false => Logger::with_env_or_str(format!("neovide = {}", verbosity)),
+        false => Logger::with_env_or_str(format!("neovide = {}", "trace")),
```
Comment (clason) [inline src/main.rs:192]: Why did you remove the `verbosity` option and force the log level to `trace`? People in general don't want the console spammed with debug output when running neovide (not to mention that this has performance implications)!

Comment (Kethku) [reply src/main.rs:192]: Does this spam to console? The logger is supposed to only write to file when the log argument is given. All other times the log calls are effectively noops

Maybe there is behavior I don't understand on some platforms?

Comment (clason) [reply src/main.rs:192]: yes, it does...

Comment (clason) [reply src/main.rs:192]: I think that's configurable; can't find out at first glance what the default is for flexi_logger

Comment (clason) [reply src/main.rs:192]: https://docs.rs/flexi_logger/latest/flexi_logger/code_examples/index.html#choose-the-log-output-channel:

> By default, logs are written to stderr.

Comment (Kethku) [reply src/main.rs:192]: Welp. I can fix this :)

Comment (clason) [reply src/main.rs:192]: There's already a PR up: https://github.com/neovide/neovide/pull/1085

Comment (Kethku) [reply src/main.rs:192]: I think the verbosity flag is unnecessary if we only log out when --log is present. That way the log always has all the information we need and never effects performance when we don't.

I think the goal originally was to always print errors to the console. Which I may still try to get going

Comment (clason) [reply src/main.rs:192]: Makes sense; you can do that pretty easily with `flex_logger`.
